ext/intl/dateformat/dateformat_parse.c

```diff
@@ -145,18 +154,59 @@ PHP_FUNCTION(datefmt_parse)
 			RETURN_FALSE;
 		}
 		parse_pos = (int32_t)long_parse_pos;
-		if((size_t)parse_pos > text_len) {
+		if ((size_t)parse_pos > text_len) {
 			RETURN_FALSE;
 		}
 	}
-	internal_parse_to_timestamp( dfo, text_to_parse, text_len, z_parse_pos?&parse_pos:NULL, return_value);
-	if(z_parse_pos) {
+	internal_parse_to_timestamp( dfo, text_to_parse, text_len, z_parse_pos ? &parse_pos : NULL, false, return_value);
+	if (z_parse_pos) {
 		zval_ptr_dtor(z_parse_pos);
 		ZVAL_LONG(z_parse_pos, parse_pos);
 	}
 }
 /* }}} */
 
+PHP_METHOD(IntlDateFormatter, parseToCalendar)
+{
+	zend_string    *text_to_parse = NULL;
+	zval*         	z_parse_pos = NULL;
+	int32_t		parse_pos = -1;
+
+	DATE_FORMAT_METHOD_INIT_VARS;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_STR(text_to_parse)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_ZVAL(z_parse_pos)
+	ZEND_PARSE_PARAMETERS_END();
+
+	object = ZEND_THIS;
+
+	/* Fetch the object. */
+	DATE_FORMAT_METHOD_FETCH_OBJECT;
+
+	if (z_parse_pos) {
+		zend_long long_parse_pos;
+		ZVAL_DEREF(z_parse_pos);
+		bool failed = false;
+		long_parse_pos = zval_try_get_long(z_parse_pos, &failed);
+		if (failed || ZEND_LONG_INT_OVFL(long_parse_pos)) {
```

Comment:
If the parsing fails, I'm pretty sure it always throws an Error so having another intl specific extension being thrown seems odd? Also a test for this case?